R/contact_matrix.r

```diff
@@ -56,8 +57,24 @@ contact_matrix <- function(survey, countries = NULL, survey.pop, age.limits, fil
   missing.participant.age <- match.arg(missing.participant.age)
   missing.contact.age <- match.arg(missing.contact.age)
 
-  ## get the survey
-  survey <- get_survey(survey)
+  error_string <-
+    "must be a survey object (created using `survey()` or `get_survey()`)"
+
+  if (all(is_doi(survey))) {
+    deprecate_warn(
+      "0.3.2", paste0("contact_matrix(survey = '", error_string, "')"),
+      details = "Passing a DOI will be removed in version 0.4.0."
+    )
+    survey <- get_survey(survey)
+  } else if (!inherits(survey, "survey")) {
+    stop(
+      "'survey' must be a survey object (created using)",
+      "or `get_survey()`)."
```

Comment:
This message was accidentally modified in the addition of the deprecation process.

```suggestion
      "'survey' must be a survey object (created using `survey()`",
      " or `get_survey()`)."
```

Reply:
addressed by using a common error string in ed8d823